code/modules/projectiles/updated_projectiles/gun_system.dm

```diff
@@ -726,104 +726,117 @@ and you're good to go.
-						if(!delete_bullet(projectile_to_fire))
-							qdel(projectile_to_fire) //If this proc DIDN'T delete the bullet, we're going to do so here.
-
-						reload_into_chamber(user) //Reload the sucker.
-
-					else
-						click_empty(user)//If there's no projectile, we can't do much.
-				else
-					M.visible_message("<span class='notice'>[user] decided life was worth living.</span>")
-				flags_gun_features ^= GUN_CAN_POINTBLANK //Reset this.
-			return
+	if(!CHECK_BITFIELD(flags_gun_features, GUN_CAN_POINTBLANK)) // If it can't point blank, you can't suicide and such.
+		return ..()
 
-		else if(user.a_intent == INTENT_HARM) //Point blanking doesn't actually fire the projectile. No reason to.
-			if(able_to_fire(user)) //If you can't fire the gun in the first place, we're just going to hit them with it.
-				if(!active_attachable && (flags_gun_features & GUN_BURST_ON) && burst_amount > 1)
-					..()
-					Fire(M, user)
-					return TRUE
-				else
-					..()
-					flags_gun_features &= ~GUN_BURST_FIRING
-					//Point blanking simulates firing the bullet proper but without actually firing it.
-					if(active_attachable && !(active_attachable.flags_attach_features & ATTACH_PROJECTILE))
-						active_attachable.activate_attachment(src, null, TRUE)//No way.
-					var/obj/item/projectile/projectile_to_fire = load_into_chamber(user)
-					if(projectile_to_fire) //We actually have a projectile, let's move on. We're going to simulate the fire cycle.
-						user.visible_message("<span class='danger'>[user] fires [src] point blank at [M]!</span>")
-						apply_bullet_effects(projectile_to_fire, user) //We add any damage effects that we need.
-						projectile_to_fire.setDir(get_dir(user, M))
-						projectile_to_fire.distance_travelled = get_dist(user, M)
-						simulate_recoil(1, user)
-
-						if(projectile_to_fire.ammo.bonus_projectiles_amount)
-							var/obj/item/projectile/BP
-							for(var/i = 1 to projectile_to_fire.ammo.bonus_projectiles_amount)
-								BP = new /obj/item/projectile(M.loc)
-								BP.generate_bullet(GLOB.ammo_list[projectile_to_fire.ammo.bonus_projectiles_type])
-								BP.setDir(get_dir(user, M))
-								BP.distance_travelled = get_dist(user, M)
-								BP.ammo.on_hit_mob(M, BP)
-								M.bullet_act(BP)
-								qdel(BP)
-
-						projectile_to_fire.ammo.on_hit_mob(M, projectile_to_fire)
-						M.bullet_act(projectile_to_fire)
-						last_fired = world.time
-
-						if(!delete_bullet(projectile_to_fire))
-							qdel(projectile_to_fire)
-						reload_into_chamber(user) //Reload into the chamber if the gun supports it.
-						if(user) //Update dat HUD
-							var/obj/screen/ammo/A = user.hud_used.ammo //The ammo HUD
-							A.update_hud(user)
-						return TRUE
-
-	return ..() //Pistolwhippin'
+	if(!able_to_fire(user))
+		return ..()
+
+	if(M != user && user.a_intent == INTENT_HARM)
+		if(!active_attachable && CHECK_BITFIELD(flags_gun_features, GUN_BURST_ON) && burst_amount > 1)
+			..()
```

Comment:
```suggestion
			. = ..()
```

Reply:
it returns a fixed result two lines below